spec/queries/customer_credit_notes_query_spec.rb

```diff
@@ -0,0 +1,145 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe CustomerCreditNotesQuery, type: :query do
+  subject(:customer_credit_notes_query) do
+    described_class.new(organization:)
+  end
+
+  let(:membership) { create(:membership) }
+  let(:organization) { membership.organization }
+  let(:customer) { create(:customer, organization:) }
+  let(:credit_note_first) { create(:credit_note, organization:, customer:, number: '11imthefirstone') }
+  let(:credit_note_second) { create(:credit_note, organization:, customer:, number: '22imthesecondone') }
+  let(:credit_note_third) { create(:credit_note, organization:, customer:, number: '33imthethirdone') }
+  let(:credit_note_fourth) { create(:credit_note, organization:, number: '44imthefourthone') }
+
+  before do
+    credit_note_first
+    credit_note_second
+    credit_note_third
+    credit_note_fourth
+  end
+
+  it 'returns all credit_notes of the customer' do
+    result = customer_credit_notes_query.call(
+      search_term: nil,
+      customer_id: customer.id,
+      page: 1,
+      limit: 10,
+    )
+
+    returned_ids = result.credit_notes.pluck(:id)
+
+    aggregate_failures do
+      expect(result.credit_notes.count).to eq(3)
+      expect(returned_ids).to include(credit_note_first.id)
+      expect(returned_ids).to include(credit_note_second.id)
+      expect(returned_ids).to include(credit_note_third.id)
+      expect(returned_ids).not_to include(credit_note_fourth.id)
+    end
+  end
+
+  context 'when searching for /imthe/ term' do
+    it 'returns only one credit_notes' do
+      result = customer_credit_notes_query.call(
+        search_term: 'imthe',
+        customer_id: customer.id,
+        page: 1,
+        limit: 10,
+      )
+
+      returned_ids = result.credit_notes.pluck(:id)
+
+      aggregate_failures do
+        expect(result.credit_notes.count).to eq(3)
```

Comment:
It looks like we are returning 3 credit notes instead of 1 :)

Reply:
Good catch ! I updated this test to search for a patttern that return 2 items that was not returned in other tests.

Udpated